packages/widget/src/controllers/wallet-manager/manager.ts

```diff
@@ -143,10 +172,7 @@ export class WalletController implements ReactiveController {
         })
       );
       if (network.chainId !== providerChainId) {
-        void provider.request({
-          method: 'wallet_switchEthereumChain',
-          params: [{ chainId: utils.hexValue(network.chainId) }]
-        });
+        await this.switchEvmChain(network.chainId);
```

Comment:
I don't think you can guarantee, provider will be set in wallet context by the time this executes. I would be better to enable passing in provider into method

Reply:
this might require us to consume the wallet context in the `FungibleTokenTransfer` component, to pass also the provider there, since `switchEvmChain` method is being called in the `onClick` handler